Session3/EFCoreAssignment.Services/ProductService.cs

```diff
@@ -0,0 +1,78 @@
+﻿using EFCoreAssignment.Data.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace EFCoreAssignment.Data.Services
+{
+    public class ProductService : IProductService
+    {
+        private readonly AppDbContext _dbContext;
+
+        public ProductService(AppDbContext dbContext) 
+        {
+            _dbContext = dbContext;
+        }
+
+        public async Task<List<ProductDto>> GetProducts()
+        {
+            // TODO get products
+            return await _dbContext.Products.Select(x => new ProductDto(x.Id, x.Name, x.ShopId)).ToListAsync();
+        }
+
+        public async Task<ProductDto> GetProduct(int id)
+        {
+            // TODO get product
+            var product = await _dbContext.Products.FirstOrDefaultAsync(x => x.Id == id) 
```

Comment:
you can just return the exact value here, service layers should not throw exceptions